ods_ci/utils/scripts/util.py

```diff
@@ -69,17 +68,16 @@ def execute_command(cmd):
             encoding="utf-8",
             errors="replace",
         ) as p:
-            while True:
-                line = p.stdout.readline()
-                if line != "":
-                    output += line + "\n"
-                elif p.poll() is not None:
-                    break
-            sys.stdout.flush()
-            log.info(f"OUTPUT: {output}")
-            return output
-    except:
-        return None
+            output = []
+            for line in p.stdout:
+                output.append(line)
+                if print_stdout:
+                    print(">:", line, end="")
+                    sys.stdout.flush()
+            return "".join(output)
```

Comment:
Why are you storing the lines in an array if then you are going to join it in a string? Is not better to store 'em directly in a string?

Reply:
Execution efficiency :slightly_smiling_face:  https://realpython.com/python-string-concatenation/#efficiently-concatenating-many-strings-with-join-in-python.

Reply:
> The .join() method is cleaner, more Pythonic, and more readable than concatenating many strings together in a loop using the augmented concatenation operator (+=), as you saw before.
You learn something new every day.
Thanks for the tip @jstourac !

Reply:
Main point for the efficiency here is this note, though :slightly_smiling_face: 

> Unlike the concatenation operators, Python’s .join() doesn’t create new intermediate strings in each iteration. Instead, it creates a single new string object by joining the elements from the input iterable with the selected separator string. This behavior is more efficient than using the regular concatenation operators in a loop.